modules/apps/portal/portal-user-locale-options/src/main/resources/META-INF/resources/dynamic_include/user_locale_options.jsp

```diff
@@ -0,0 +1,60 @@
+<%--
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ include file="/dynamic_include/init.jsp" %>
+
+<%
+Locale userLocale = user.getLocale();
+%>
+
+<liferay-util:buffer
+	var="alertMessage"
+>
+	<div dir="<%= LanguageUtil.get(userLocale, "lang.dir") %>">
+		<div class="d-block">
+			<%= LanguageUtil.format(userLocale, "this-page-is-displayed-in-x", locale.getDisplayName(userLocale)) %>
+		</div>
+
+		<c:if test="<%= LanguageUtil.isAvailableLocale(themeDisplay.getSiteGroupId(), user.getLocale()) %>">
+			<aui:a cssClass="d-block" href='<%= themeDisplay.getPathMain() + "/portal/update_language?p_l_id=" + themeDisplay.getPlid() + "&redirect=" + URLCodec.encodeURL(currentURL) + "&languageId=" + user.getLanguageId() + "&persistState=false&showUserLocaleOptionsMessage=false" %>'>
+				<%= LanguageUtil.format(userLocale, "display-the-page-in-x", userLocale.getDisplayName(userLocale)) %>
+			</aui:a>
+		</c:if>
+	</div>
+
+	<div dir="<%= LanguageUtil.get(request, "lang.dir") %>">
+		<aui:a cssClass="d-block" href='<%= themeDisplay.getPathMain() + "/portal/update_language?p_l_id=" + themeDisplay.getPlid() + "&redirect=" + URLCodec.encodeURL(currentURL) + "&languageId=" + themeDisplay.getLanguageId() + "&showUserLocaleOptionsMessage=false" %>'>
+			<%= LanguageUtil.format(locale, "set-x-as-your-preferred-language", locale.getDisplayName(locale)) %>
+		</aui:a>
+	</div>
+</liferay-util:buffer>
+
+<aui:script>
+	Liferay.Util.openToast({
+		message: '<%= HtmlUtil.escapeJS(alertMessage) %>',
+		messageType: 'html',
+		onClose: function (data) {
+			if (data.event) {
+				Liferay.Util.Session.set('ignoreUserLocaleOptions', true);
+				Liferay.Util.Session.set('useHttpSession', true);
+			}
+
+			data.unmount();
```

Comment:
I know it's too late to say this given I already gave my approval, but: This looks like could be encapsulated by the component itself. Although, there might be cases where devs wouldn't want to unmount right away (like only unmount after an async request is done). Is that why you exposed the `data.unmount`?

Reply:
That's what I thought, yeah... that's what we do in the `onClick` as well... we pass the `onClose` method, which calls `unmount`. I imagine this as like the `event.preventDefault` version in React...

Reply:
Yeah, for `onClick` it makes more sense, because devs can choose to close or not the toast on clicks.

Reply:
But that's not a big deal. I'm just afraid people might forget to unmount, but I'm guessing this would be an obvious mistake to correct given the if one forgets to unmount it wont actually get dismissed, right? Or do we hide it with CSS?

Reply:
No, if you don't unmount the toast will stay, so yeah, it should be pretty obvious.

I wouldn't mind changing this either, so let's wait for @wincent or @bryceosterhaus or anyone else to chime in?

Reply:
I don't mind this going in like it is, but my personal impression is that if you're in `onClose()` its because the user explicitly signalled their intention to dismiss the toast in some way (by taking action), or the developer did (by setting `autoClose`), so it is "too late" at that point to expect the toast to stick around. A toast is not like a complex screen that you would want to guard against accidental navigation (and loss of state); it's supposed to be a simple, ephemeral, disposable notification.

So I don't think I'd pass in `unmount`; I'd just call automatically.

Reply:
I don't have a super strong preference, but I would lean towards Greg's argument.

Reply:
Lol, yeah, I totally agree 👍